src/test/java/com/squareup/kotlinpoet/PropertySpecTest.kt

```diff
@@ -95,4 +95,28 @@ class PropertySpecTest {
       |external val foo: kotlin.String
       |""".trimMargin())
   }
+
+  @Test fun initializer() {
+    val prop = PropertySpec.builder("foo", String::class)
+            .initializer("\"bar\"")
+            .build()
+
+    assertThat(prop.toString()).isEqualTo("val foo: kotlin.String = \"bar\"\n")
+  }
+
+  @Test fun initializerWithTypeInferrence() {
+    val prop = PropertySpec.builder("foo", String::class)
+        .initializerWithTypeInference("\"bar\"")
+        .build()
+
+    assertThat(prop.toString()).isEqualTo("val foo = \"bar\"\n")
+  }
+
+  @Test fun delegateWithTypeInferrence() {
+    val prop = PropertySpec.builder("foo", String::class)
+            .delegateWithTypeInference("Delegates.notNull<String>()")
```

Comment:
This violates the guarantees otherwise provided by the library. A sibling or nested class with the name `Delegates` or `String` will break this.